src/HalResource.php

```diff
@@ -42,20 +42,25 @@ class HalResource implements EvolvableLinkProviderInterface, JsonSerializable
     /** @var array<array-key, self|array<array-key, self>> */
     private $embedded = [];
 
+    private bool $embedEmptyCollections;
+
     /**
      * @param array $data
      * @param LinkInterface[] $links
      * @param HalResource[][] $embedded
      */
-    public function __construct(array $data = [], array $links = [], array $embedded = [])
-    {
+    public function __construct(
+        array $data = [],
+        array $links = [],
+        array $embedded = [],
+        bool $embedEmptyCollections = false
```

Comment:
We can use constructor property promotion here:

```php
private bool $embedEmptyCollections = false
```

That will allow eliminating the property declaration, as well as the assignment on line 58.